app/models/mentee_application_validation.rb

```diff
@@ -93,10 +93,16 @@ def checked?(value)
       optional(:other_programming_languages).value(:array, size?: 1..4).each(:filled?)
       required(:previous_participation).filled(:bool?)
       optional(:perceived_methods).each(:filled?)
+      optional(:programming_experience) { array? }
+      optional(:programming_experience_level) { filled? }
     end
 
     rule(:gdpr_consent) do
       key.failure(:unchecked) unless checked?(value)
     end
+
+    rule(:programming_experience).each do
+      base.failure('Programming level of expertise must have a value') if values.data[:previous_programming_experience] && values.data[:programming_experience_level][value].nil?
+    end
```

Comment:
Currently after making it mandatory, it's not throwing error if I click next. So it seems page is hanged. So user will not know why they can't move next.
Also I have following 2 qs:
Also are we making mandatory for all languages ?
It's necessary only when some select the language right ?

Reply:
Yes, only the selected languages must have a expertise value. I pushed the fix for page hanging issue.